File: 《亿级流量Jαva高并发与网络编程实战》-源码/demo/ch08/thrift/python/StudentServiceImpl.py
#以python方法导入thrift相关类
from  genpy.thrift.generatecode import  ttypes

class StudentServiceImpl:
        #实现查询方法
        def queryStudents(self):
            print("-----Python服务端，模拟查询操作------")
            student1 = ttypes.Student()
            student1.name = 'zs'
            student1.age = 23

            student2 = ttypes.Student()
            student2.name = 'ls'
            student2.age = 24

            list = [student1,student2]
            print("-----查询完毕------")
            return list

        # 实现增加方法
        def saveStudent(self,name,age):
            print("-----Python服务端,模拟增加操作------")
            print ("姓名:%s，年龄:%d" % (name ,age) )
            print("-----模拟成功------")
            return True

File: 《亿级流量Jαva高并发与网络编程实战》-源码/demo/ch08/thrift/StudentServiceImpl.java
package com.yanqun.thrift;

import com.yanqun.thrift.generatecode.MyException;
import com.yanqun.thrift.generatecode.Student;
import com.yanqun.thrift.generatecode.StudentService;
import org.apache.thrift.TException;

import java.util.ArrayList;
import java.util.List;

public class StudentServiceImpl implements StudentService.Iface {
    @Override
    public List<Student> queryStudents() throws TException {
        System.out.println("--Java服务端，模拟查询操作--");
        Student Student1 = new Student();
        Student1.setName("zs") ;
        Student1.setAge(23) ;

        Student Student2 = new Student();
        Student2.setName("ls") ;
        Student2.setAge(24) ;

        List<Student> Students = new ArrayList<>();
        Students.add(Student1);
        Students.add(Student2);
        System.out.println("--查询完毕--");
        return Students;
    }

    @Override
    public boolean addStudent(String name, int age) throws MyException, TException {
            System.out.println("--Java服务端，模拟增加操作--");
            System.out.println("增加成功：" +name+","+age);
            return true;
    }



}
